### PR TITLE
interop explainers

### PR DESCRIPTION
**Pull request by cpengilly**
2 interop explainers

### CODE REVIEW

**Review by sbvegan — COMMENTED** [review]
This looks great!

File: pages/stack/protocol/interop/cross-chain-message.mdx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+lang: en-US
+description: Learn how cross-chain messaging works with OP Stack interoperability.
+---
+
+import { Callout } from 'nextra/components'
+import Image from 'next/image'
+
+# Anatomy of a Cross-Chain Message
+
+A cross-chain message applies to any asset using the [SuperchainERC20](https://specs.optimism.io/interop/token-bridging.html) token standard. Two transactions are required to complete a cross chain message:
```
Comment (zainbacchus) [inline pages/stack/protocol/interop/cross-chain-message.mdx:12]: A cross-chain message applies to any message sent across chain. This applies to asset transfers using the SuperchainERC20 token standard.

File: pages/stack/protocol/interop/cross-chain-message.mdx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+*   **First transaction:** is submitted to the source chain and emits an event that can be consumed on a destination chain.
+*   **Second transaction:** is submitted to a destination chain, where the block builder SHOULD only include it if certain that the first transaction was included in the source chain. NOTE: There is no strict requirement that the executing message is ever submitted.
```
Comment (zainbacchus) [inline pages/stack/protocol/interop/cross-chain-message.mdx:15]: Anyone can submit the second transaction. The block builder can use OP-Supervisor to determine the integrity of the initating message. 

cc @tynes

File: pages/stack/protocol/interop/cross-chain-message.mdx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+<Callout type="info">
+  See the specs for details on tracing the [executing message event](https://specs.optimism.io/interop/predeploys.html#executingmessage-event).
+</Callout>
+
+## How It Works
```
Comment (zainbacchus) [inline pages/stack/protocol/interop/cross-chain-message.mdx:21]: (nit) Shouldnt this section be first?

File: pages/stack/protocol/interop/explainer.mdx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+title: Interoperability Explainer
+lang: en-US
+description: Learn the basics of interoperability on the OP Stack.
+---
+
+import { Callout } from 'nextra/components'
+
+# Interoperability Explainer
+
+Interoperability is the ability for a blockchain to easily read the state of another blockchain.
```
Comment (zainbacchus) [inline pages/stack/protocol/interop/explainer.mdx:11]: easily => securely

File: pages/stack/protocol/interop/explainer.mdx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+This means ETH and ERC-20s can seamlessly and securely move across L2s, and intent-based protocols, such as bridges, can build better experiences on top of the message passing protocol.
+
+## Low Latency
+Low latency interoperability allows for horizontally scalable blockchains, a key feature of the Superchain. With native interop, latency can be low (~2 seconds) by optimistically accepting cross chain messages. 
+The fork choice rule enforces eventual consistency, meaning that if an invalid cross chain message is accepted, it will be reorganized out eventually. 
+The fault proof guarantees that all of the cross chain messages are accounted for from the perspective of handling withdrawals through the bridge to L1.
+
+## Permissionless Chain Set
+It is permissionless to define a dependency on a chain, so chain operators will be able to choose which chains their chains depend on, and it is not a requirement that chains are in each other's dependency set.
```
Comment (zainbacchus) [inline pages/stack/protocol/interop/explainer.mdx:32]: The dependency set for OP Mainnet will be managed by Optimism Governance. 

cc @tynes

File: pages/stack/protocol/interop/explainer.mdx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+## Permissionless Chain Set
+It is permissionless to define a dependency on a chain, so chain operators will be able to choose which chains their chains depend on, and it is not a requirement that chains are in each other's dependency set. 
+Chain operators can add or remove chains from the dependency set through the `SystemConfig`. For example, Optimism Governance is the arbiter of the dependency set of OP Mainnet.
+
+However, since the nature of defining a dependency is one way, it is impossible for a chain to know of all of the other chains that depend on it.
+
+## Considerations
+Chain operators will need to run additional infrastructure to be part of the interoperable set. 
+*   The Superchain-backend `microservice` will be a requirement for running an OP Stack chain that has interop enabled.
```
Comment (zainbacchus) [inline pages/stack/protocol/interop/explainer.mdx:39]: isn't microservice OP-Supervisor? cc @tynes

File: pages/stack/protocol/interop/explainer.mdx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+It is permissionless to define a dependency on a chain, so chain operators will be able to choose which chains their chains depend on, and it is not a requirement that chains are in each other's dependency set. 
+Chain operators can add or remove chains from the dependency set through the `SystemConfig`. For example, Optimism Governance is the arbiter of the dependency set of OP Mainnet.
+
+However, since the nature of defining a dependency is one way, it is impossible for a chain to know of all of the other chains that depend on it.
+
+## Considerations
+Chain operators will need to run additional infrastructure to be part of the interoperable set. 
+*   The Superchain-backend `microservice` will be a requirement for running an OP Stack chain that has interop enabled. 
+This service is responsible for validating all cross chain messages and will need to have an RPC configured for each chain in the dependency set. 
+*   In the future, this service will rely on the P2P network and cryptographic schemes for validating cross chain messages, instead of RPCs.
```
Comment (zainbacchus) [inline pages/stack/protocol/interop/explainer.mdx:41]: In the future, to reduce the infrastructure cost, this service will...

cc @tynes

File: pages/stack/protocol/interop/explainer.mdx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+*   For more info about how OP Stack interoperability works under the hood, [check out the specs](https://specs.optimism.io/interop/overview.html).
+
+## FAQs
+
+### What is the latency/security tradeoff?
+If a sequencer does not want to trust another sequencer at all, they can choose to only accept executing messages where the initiating message has been finalized with L1 levels of security. This demonstrates the difference in latency vs speed. The L2 data must be submitted to L1 and then the L1 block containing the transaction with the L2 data must be finalized. It takes about 15 minutes for an L1 block to finalize, and higher throughput OP Stack chains like Base and OP Mainnet submit a batch about every 5 minutes. This means that it could take about 20 minutes for an L2 transaction to be considered final and be able to be trustlessly consumed by another chain. At lower latencies, the sequencer accepting the executing message takes on some amount of equivocation risk. It could be possible to build a bonding system to add economic security to prevent equivocation, but that is not in the scope of the first release.
```
Comment (zainbacchus) [inline pages/stack/protocol/interop/explainer.mdx:55]: should we add the safe/unsafe diagram here? cc @sbvegan

**Review by zainbacchus — CHANGES_REQUESTED** [review]

File: pages/stack/protocol/interop/explainer.mdx (hunk 8 above)
Comment (sbvegan) [reply pages/stack/protocol/interop/explainer.mdx:55]: yeah I think its a good place for it, @cpengilly here's the image we can use to illustrate this point

unsafe-safe-finalize

File: pages/stack/protocol/interop/explainer.mdx (hunk 6 above)
Comment (cpengilly) [reply pages/stack/protocol/interop/explainer.mdx:40]: confirmed with @tynes that it is op-supervisor, so updating docs now 😅

File: pages/stack/protocol/interop/cross-chain-message.mdx [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+A cross-chain message applies to any message sent across a chain. This applies to asset transfers using the [SuperchainERC20](https://specs.optimism.io/interop/token-bridging.html) token standard. 
+
+## How It Works
+
+To send a cross-chain message on the Superchain using [native OP Stack interoperability](explainer), these two aspects must be in place:
+
+1.  Each interoperable chain runs a verifying node for each chain in the interoperable set.
+2.  Each cross-chain message has an **initiating transaction** on the source chain and a **finalizing transaction** on the destination chain.
+    *   **First/initiating transaction:** is submitted to the source chain and emits an event that can be consumed on a destination chain.
+    *   **Second/finalizing transaction:** is submitted to a destination chain, where the block builder SHOULD only include it if certain that the first transaction was included in the source chain. The block builder can use OP-Supervisor to determine the integrity of the initiating message. Anyone can submit the second transaction.
```
Comment (bradleycamacho) [inline pages/stack/protocol/interop/cross-chain-message.mdx:21]:
```suggestion
    *   **Second/finalizing transaction:** is submitted to a destination chain, where the block builder should only include it if certain that the first transaction was included in the source chain. The block builder can use OP-Supervisor to determine the integrity of the initiating message. Anyone can submit the second transaction.
```

Comment (bradleycamacho) [reply pages/stack/protocol/interop/cross-chain-message.mdx:21]: Capitalizing `SHOULD` seemed weird. Doesn't seem to need emphasis, and if the implication is "it might or might not", we should probably be a bit more clear

File: pages/stack/protocol/interop/explainer.mdx [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+*   less fragmentation across the Superchain
+*   improved user experience for developers on the Superchain
+
+## Secure Message Passing
+Native interop includes both the protocol layer message passing and the Superchain ERC20 token specification.
+
+*   **Message passing protocol:** the initial + finalizing/executing [message](https://specs.optimism.io/interop/messaging.html) that fire events to be consumed by the chains in the [dependency set](https://specs.optimism.io/interop/dependency-set.html)
+*   **SupERC20 token specification**: the [SuperchainERC20](https://specs.optimism.io/interop/token-bridging.html) turns message passing into asset transfer between chains in the interop set
+
+This means ETH and ERC-20s can seamlessly and securely move across L2s, and intent-based protocols, such as bridges, can build better experiences on top of the message passing protocol.
```
Comment (bradleycamacho) [inline pages/stack/protocol/interop/explainer.mdx:25]: This sentence was pretty hard to parse. Not a blocker, just wanted to highlight.

File: pages/stack/protocol/interop/explainer.mdx [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+
+## Low Latency
+Interoperability allows for horizontally scalable blockchains, a key feature of the Superchain. With native interop, latency can be low (~2 seconds) by optimistically accepting cross chain messages.
```
Comment (bradleycamacho) [inline pages/stack/protocol/interop/explainer.mdx:28]: Should all instances of `cross chain` be `cross-chain`?

File: pages/stack/protocol/interop/explainer.mdx [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+This means ETH and ERC-20s can seamlessly and securely move across L2s, and intent-based protocols, such as bridges, can build better experiences on top of the message passing protocol.
+
+## Low Latency
+Interoperability allows for horizontally scalable blockchains, a key feature of the Superchain. With native interop, latency can be low (~2 seconds) by optimistically accepting cross chain messages. 
+The fork choice rule enforces eventual consistency, meaning that if an invalid cross chain message is accepted, it will be reorganized out eventually. 
+The fault proof guarantees that all of the cross chain messages are accounted for from the perspective of handling withdrawals through the bridge to L1.
+
+## Permissionless Chain Set
+It is permissionless to define a dependency on a chain, so chain operators will be able to choose which chains their chains depend on, and it is not a requirement that chains are in each other's dependency set. 
+Chain operators can add or remove chains from the dependency set through the `SystemConfig`. For example, the dependency set for OP Mainnet is managed by Optimism Governance.
```
Comment (bradleycamacho) [inline pages/stack/protocol/interop/explainer.mdx:34]: Do we need an example of this?

File: pages/stack/protocol/interop/explainer.mdx [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+
+### Are callback style transactions possible?
+If two blocks are being built at the same time with shared knowledge of their contents, it is possible to build blocks where a transaction calls to another chain, does compute and then a transaction calls back with the results. This requires no protocol level changes, it just requires more sophisticated block builder infrastructure.
+
+### What is stopping a shared sequencer from including just the executing message and not the initiating message?
+The protocol enforces the fact that all executing messages are valid. It does this by reorganizing out executing messages that have invalid initiating messages. Running software that does not enforce this would be non-standard behavior and would leave yourself at risk of accepting an invalid executing message and therefore running on a forked chain.
+
+### What is the trust/verification model? Do Sequencers that opt into this interop system have to trust each other fully?
```
Comment (bradleycamacho) [inline pages/stack/protocol/interop/explainer.mdx:73]:
```suggestion
### What is the trust/verification model? Do sequencers that opt into this interop system have to trust each other fully?
```

File: pages/stack/protocol/interop/explainer.mdx (hunk 12 above)
Comment (cpengilly) [reply pages/stack/protocol/interop/explainer.mdx:34]: yep, it's in the specs :)

File: pages/stack/protocol/interop/explainer.mdx (hunk 11 above)
Comment (cpengilly) [reply pages/stack/protocol/interop/explainer.mdx:28]: updated, we should prob add this to linter too, which I'll do in a separate PR

**Review by zainbacchus — APPROVED** [review]
lgtm
